### DISCUSSION TITLE
Sujet / Thema 6 –– Feedback-Loop für Geodaten / Circuit de feedback pour les géodonnées

### DISCUSSION THREAD

**Discussion by rastrau** [GeoUnconference/discussions#5] [Ideen / Idées]
Resultate der Diskussion an der GeoUnconference 1 / Résultats de la discussion lors de la GeoUnconference 1

![Feedback loop for geodata](https://user-images.githubusercontent.com/4510815/124478661-b6bb4280-dda5-11eb-9eb3-41fc4f353c7f.jpg)

**Comment by davidoesch** [comment]
@RafBov hast du hier noch wietere Ideen für Umsetzungen aus deiner Erfahrung mit revisionsdienst swisstopo?

**Comment by nrohrbach** [comment]
Bei ichtankestrom.ch und bei sharedmobility.ch ist ein Feedbacklink direkt im Tooltip verfügbar: https://s.geo.admin.ch/91c61aa130
zB: https://www.uvek-gis.admin.ch/BFE/disharing/feedback/DISharing_Feedback/?name=Samnaun+-+Hangl%60s+Sport+%26+Mode&provider=Rent+a+Bike&lang=de&lon=10.3611&lat=46.9441

Sehr viele Rückmeldungen kommen über sonnendach.ch rein wo das Feedback-Widget von swisstopo prominent platziert ist.

**Reply by rastrau** [reply]
Spannend! Dazu würden mich einige Fragen interessieren:
1. Wie viele Feedbacks gehen pro Jahr circa ein?
2. Wie sind die Typen von Feedbacks verteilt: z.B. zur Darstellung, zu (falscher) Lage, zu (falschen) Attributen, zu fehlenden Objekten, zu nicht mehr vorhandenen Objekten u.ä.?
3. Wie werden die Feedbacks beim BFE bearbeitet? 
4. Gibt es eine Rückmeldung an die Befundgeber(innen) und/oder Transparenz darüber, wo im Prozess das Feedback steht?
5. Gibt es eine Rückmeldung an die Befundgeber(innen) was final durch das Feedback umgesetzt wurde?

**Reply by nrohrbach** [reply]
1. Von sonnendach.ch erhalten wir 2-3 Feedbacks pro Woche. Bei den anderen Layern 2-3 Feedbacks pro Monat.
2. Bei sonnendach.ch primär fehlende Objekte bzw. nicht parzellenscharfe Objekte. sonst in der Regel falsche Attributierung oder falsche Lage. Manchmal wird auch einfach die Kontaktaufnahme gesucht.
3. Ich prüfe die Meldungen und wenn möglich werden die Daten für das nächste Update korrigiert bzw. die zuständigen Stellen informiert.
4. Jede nicht-anonyme Rückmeldung wird beantwortet. 
5. Eine finale Meldung gibt es nicht. Korrekturen werden wenn möglich sofort umgesetzt.

**Reply by rastrau** [reply]
Sehr cool. Aus Nutzendensicht finde ich es wertvoll und motivierend, wenn ich erfahre, was mit meiner Rückmeldung geschieht. Wenn der interne Datenworkflow so ist, dass Rückmeldungen sehr schnell verarbeitet werden können (Stunden bis Tage oder maximal Wochen), umso besser. 

Gleichzeitig finde ich, soll auch ein Opt-Out möglich sein, so dass ich eine Rückmeldung machen und gleichzeitig angeben kann, dass ich nicht mehr über die weiteren Schritte informiert werden möchte.

**Comment by rastrau** [comment]
(via @hinjud) Für Services der Kantone auf geodienste.ch (und eigentlich auch für die gesamte Website) gibt es ein anonymes Feedback-Formular: https://geodienste.ch/feedback Nutzer(innen) können freiwillig die Art ihrer Institution angeben. 

Kontaktangaben der Befundmelder(innen) sind nicht vorgesehen. Rückmeldungen zur Feedback-Bearbeitung dürften damit ausbleiben, ausser man gibt seine Mail-Adresse im Befund mit ein (?).

Auch hier fände ich interessant:
1. Wie viele Feedbacks gehen pro Jahr circa ein?
2. Wie sind die Typen von Feedbacks verteilt: z.B. zur Website, zu Datensätzen/Services, zu (falscher) Lage, zu (falschen) Attributen, zu fehlenden Objekten, zu nicht mehr vorhandenen Objekten u.ä.?
3. Wie werden die Feedbacks bei KGK / dem betroffenen Kanton bearbeitet?
4. Gibt es aus Sicht KGK Verbesserungspotenzial bei den Feedbacks?

**Reply by dkottmann** [reply]
@rastrau 

> Wie viele Feedbacks gehen pro Jahr circa ein?

Auf geodienste.ch steht das Feedbackformular erst seit dem 1.4.2021 zur Verfügung. 

> Wie sind die Typen von Feedbacks verteilt: z.B. zur Website, zu Datensätzen/Services, zu (falscher) Lage, zu (falschen) Attributen, zu fehlenden Objekten, zu nicht mehr vorhandenen Objekten u.ä.?

Siehe oben. 

> Wie werden die Feedbacks bei KGK / dem betroffenen Kanton bearbeitet?

* Für Fehlermeldungen oder Supportanfragen verwenden die Kunden unser Supportformular resp. das Ticketsystem oder melden sich direkt per E-Mail. Dies ist nicht anonym und gibt uns die Möglichkeit, die Anfragen innert kurzer Frist zu bearbeiten/beantworten. 
* Feedbacks via Feedbackformular sind anonym und werden nicht beantwortet. Kontrolle der Feedbacks erfolgt mind. alle zwei Wochen. Verbesserungsmöglichkeiten werden in den Change Prozess aufgenommen. Fehlermeldungen landen bisher eher selten im Feedbackformular. Falls doch: ggf. werden Fehlermeldungen ins Support Ticketsystem übertragen und intern bearbeitet (falls nicht durch Kunde bereits geschehen)
* Zusätzlich werden die Feedbacks dreimal jährlich durch das Kommunikationsteam überprüft. Dabei werden weitere Verbesserungsmöglichkeiten reflektiert / diskutiert und ggf. in den Change Prozess aufgenommen. 

> Gibt es aus Sicht KGK Verbesserungspotenzial bei den Feedbacks?

Ich könnte mir folgendes vorstellen: 
* Evtl. wäre es interessant für die Kunden, wenn die Feedbacks nach einer internen Überprüfung publiziert würden
* Zusätzlich könnten wir Hinweise / Bemerkungen ergänzen, beispielsweise wenn eine Verbesserung umgesetzt wurde

**Reply by rastrau** [reply]
Vielen Dank für diese Angaben und auch die weiterführenden Hinweise!

**Reply by davidoesch** [reply]
https://twitter.com/swiss_geoportal/status/1439987590715822089

![image](https://user-images.githubusercontent.com/4577727/134037809-629d811b-40ca-433c-97ab-b4502fb67549.png)

E-mail inbox of 
@swiss_geoportal
: ~20'000 mails in 4 yrs, avg ~13/day, 1/4 is about map/data errors, 3/4 general inquiries, only 2% anonymou

**Comment by davidoesch** [comment]
Für geo.admin.ch

1. Wie viele Feedbacks gehen pro Jahr circa ein?  -> Pro Tag zwischen 5-20 mail (und ca 2 Fragen via https://groups.google.com/g/geoadmin-api pro Woche)

2. Wie sind die Typen von Feedbacks verteilt: z.B. zur Website, zu Datensätzen/Services, zu (falscher) Lage, zu (falschen) Attributen, zu fehlenden Objekten, zu nicht mehr vorhandenen Objekten u.ä.? -> Von 10 Fragen sind 4 zu "Daten" (swisstopo Daten Kartenmeldung, BFS Adressen und andere Bundesämter) , Nutzung (öppis tuet nid): Dienst oder Kartenviewer, und dann allgemeine INFO , wie Datenbestellen

3. Wie werden die Feedbacks bei swisstopo / dem betroffenen Kanton bearbeitet?: -> mail innerhalb Arbeitstag beantworten

**Comment by rastrau** [comment]
Die Gruppe hat folgende **Lösungsskizze** vorgestellt:

![image](https://user-images.githubusercontent.com/4510815/133230570-161f2ebd-f386-4490-9763-fb5fc5a67e5d.png)

... und für das **initiale Brainstorming** / die **Problemerfassung** für den Feedbackloop folgende Notizen gemacht:

![IMG_20210909_161545](https://user-images.githubusercontent.com/4510815/133229977-dfe64455-2b16-43ce-9c3e-8ff035d4bf09.jpg)

**Reply by schenardi** [reply]
Was ich an der Konferenz selber eigentlich noch hatte fragen wolle (die Zeit fehlte leider): 
Sollte nicht auch der umgekehrte Loop verbessert werden? 

**Wie erfahren Datenbenutzer, dass sich an den Daten selber etwas verändert wird oder verändert hat?**
Wie kann das maschinen-nutzbar bereitgestellt werden ohne dass stets mittels Stac-API jeder Timestamp abgefragt werden muss?

Anforderung ist z.B. die, dass ich meine selbst gehosteten Basisdienste (z.B. Höhenmnodell) mit den jeweils aktuellsten Daten aktualisieren will oder auch Zeitreihen zeitnah bereitstellen könnte.

**Reply by davidoesch** [reply]
einen PUSH Service gibt es nicht ( inwiefern wir SNS https://aws.amazon.com/de/sns/ auf data.geo.admin.ch aktivieren werden, der PUSH service anbietet--das ist noch offen)

Das best practice Vorgehen

gemäss Beispiel "Abfrage der Items des Datensatzes swissBUILDINGS3D 2.0 innerhalb eines bestimmten Datumsbereichs"
in
https://www.geo.admin.ch/de/geo-dienstleistungen/geodienste/downloadienste/stac-api.html#example

abfragen für welchen Datensatz im Zeitraum  X bis X1 neue Daten vorliegen

`https://data.geo.admin.ch/api/stac/v0.9/collections/<collection>/items?datetime=<Datum von>/<Datum bis>`

zb

`https://data.geo.admin.ch/api/stac/v0.9/collections/ch.swisstopo.swissbuildings3d_2/items?datetime=2018-01-01/2018-12-31`


Resultat: Liste der entsprechenden items die neu sind, mit entsprechenden links zu den assets zb swissbuildings3d_2_2018-07_1064-24_2056_5728.dxf.zip

**Reply by rastrau** [reply]
Advanced. 

Eine einfache Daten-Update-Subskription auf gängigen / gängigeren Kanälen (E-Mail, RSS, ...?) wäre für manche Geodaten und Nutzerinnen und Nutzer sicher wertvoll.

**Comment by davidoesch** [comment]
RSS really ? Wir haben für geo.admin.ch https://api3.geo.admin.ch/releasenotes/rss2.xml und die News group mit dem announce Keyword im Subject http://groups.google.com/group/geoadmin-api

auf Feature Ebene (eg update einer swissimage Kachel) haben wir interne Notifikation—- diese könnte man veröffentlichen

**Reply by rastrau** [reply]
"Really?": Bist Du überrascht den Kanal-Wunsch oder darüber, dass der API-RSS nicht allgemein bekannt ist? Man lernt nie aus an der GeoUnconference 🙂
 
Ich würde beim API-RSS spontan keine "Daten-News" suchen, ausser ev. Änderungen am Datenmodell u.ä. (nicht z.B. Änderungen an Inhalten, oder höchstens summarisch). Der Newsgroup-"Trick" scheint mir auch etwas "hacky".

Mir kamen noch die News im geowebforum zur BGDI in den Sinn, aber die sind kaum nutzbar (jedenfalls nicht anhand des Thread-Titels schnell scannbar, da stets am selben Ort gepostet).

Was existiert bezüglich @schenardi's Frage bei den Kantonen? Bei den Städten/Gemeinden (wohl nichts/wenig)? Bei der Aggregationsinfrastruktur der Kantone (geodienste.ch) gibt es einen Newsletter und ein Changelog, das z.T. auch sehr kompakt Datenänderungen enthält.

**Reply by davidoesch** [reply]
RSS: für Datennews wäre eigentlich ATOM/OPENSEARCH wie @dkottmann unten beschrieben hat das tool of choice 

Weitere Möglichkeiten

- Aus dem STAC API liese sich einfach ein Dashboard analog zu https://data.maptiler.com/changelog/?all=True aufbauen den STAC lässt auch neben datetime  auch BBOX Abfragen zu , zb https://data.geo.admin.ch/api/stac/v0.9/collections/ch.swisstopo.swissbuildings3d_2/items?datetime=2018-01-01/2018-12-31&bbox=6.43,42.95,8.69,49.10
-  Das RESTAPI verfügt mit http://api.geo.admin.ch/services/sdiservices.html#layers-metadata ebenfalls die Möglichkeit den Datenstand zu pollen 
- Geocat CSW enthält ebenfalls ein Feld für den Datenstand

Fazit: die Dienste für ein "Schweizweites Datenstanddashboard" wäre vorhanden

**Reply by hinjud** [reply]
@davidoesch Ich fände es super, wenn dies so gemacht würde. 

> auf Feature Ebene (eg update einer swissimage Kachel) haben wir interne Notifikation—- diese könnte man veröffentlichen

**Comment by dkottmann** [comment]
> Was existiert bezüglich @schenardi's Frage bei den Kantonen? Bei den Städten/Gemeinden (wohl nichts/wenig)? Bei der Aggregationsinfrastruktur der Kantone (geodienste.ch) gibt es einen Newsletter und ein Changelog, das z.T. auch sehr kompakt Datenänderungen enthält.

Auf geodienste.ch gibt es den eCH-0056 Download-Dienst (AtomFeed + OpenSearch): https://geodienste.ch/atom/download.xml 

Der Feed kann mit einem beliebigen RSS Reader gelesen werden und beispielsweise für automatisierte Datenintegrationen verwendet werden. Enthalten sind gemäss eCH-0056 die INTERLIS Daten. Wäre aber einfach erweiterbar für andere Formate.

**Reply by hinjud** [reply]
Beim Kanton Bern werden die Änderungen je nach betroffener Thematik vorab und/oder kurz nach Publikation als Aktuellmeldung auf dem Geoportal publiziert. Diese kann als RSS-Feed bezogen werden.

https://www.geo.apps.be.ch/de/component/obrss/aktuelle-meldungen.feed

Jedoch mussten wir feststellen, dass die Kommunikation via RSS-Feed etc. allein die Bedürfnisse der Nutzer:innen nicht abdeckt. Es ist daher angedacht, dass es in Zukunft die Möglichkeit geben soll, via Geoportal eine Art "Benachrichtigungsprofil" zu erfassen, um spezifisch über Aktualisierungen etc. für jene Geodaten/Dienste (wenn möglich bis auf Stufe Ebene) informiert zu werden, die für den/die jeweilige/n Nutzer:in von Relevanz sind. So haben Nutzer:innen beispielsweise auch die Möglichkeit Applikationen so vorzubereiten, dass geplante Datenmodelländerungen reibungslos nachvollzogen werden können (v.a. für geodienstbasierten Datenbezug relevant).

**Comment by rastrau** [comment]
An der GeoUnconference 2 haben wir übrigens noch den Standard 311 erwähnt. Hierzu findet man unter [open311.org](http://www.open311.org) bzw. [wiki.open311.org](http://wiki.open311.org) mehr Informationen. Der Standard ist übrigens nach einer [in den USA und Kanada gebräuchlichen Telefonnummer](https://en.wikipedia.org/wiki/3-1-1) (ähnlich 911) benannt. 

Ein relativ bekanntes Projekt, das den Standard nutzt (allerdings für *real-world issues* wie kaputte Strassenlaternen u.ä.) ist [FixMyStreet](https://www.fixmystreet.com/). Soweit ich weiss, basiert die Beschwerde-/Melde-App [Züri Wie Neu](https://www.zueriwieneu.ch/) auf FixMyStreet. Ich habe noch nie untersucht, ob/wie gut dieser Standard sich allenfalls auch auf (Geo)Daten (oder gar «side-matter» wie ein Geoportal, Webauftritt o.ä.) anwenden liesse.

**Comment by davidoesch** [comment]
Bei der Umsetzung Sperrungen und Umleitungen von Wanderwegen sowie Velo- und Mountainbikerouten via Dienst https://www.geo.admin.ch/de/home.detail.news.html/geo-internet/news2020/news20200820.html haben wir  311 / Georeport http://wiki.open311.org/GeoReport_v2/ als möglichen Kandidaten evaluiert. Da die Weiterentwiklung des Georeport unklar war und IMHO dies auch in eCH / OGC etabliert werden müsste --haben wir uns auf einen reinen RESTful API limitiert -vorerst
